pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -63,8 +66,10 @@ func (s objectReference) String() string {
 // GarbageCollector.processItem() reads the nodes, but it only reads the fields
 // that never get changed by Propagator.processEvent().
 type node struct {
-	identity   objectReference
-	dependents map[*node]struct{}
+	identity objectReference
+	// dependents will be read by the orphan() routine, we need to protect it with a lock.
+	dependentsLock *sync.RWMutex
```

Comment:
optional nil: as long as nodes are always passed by pointer, this doesn't need to be a pointer.